airflow/contrib/kubernetes/pod_launcher.py

```diff
@@ -0,0 +1,103 @@
+
+# -*- coding: utf-8 -*-
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import json
+import time
+from airflow.utils.log.logging_mixin import LoggingMixin
+from airflow.utils.state import State
+from kubernetes import watch
+from kubernetes.client.rest import ApiException
+
+from .kube_client import get_kube_client
+
+
+class PodStatus(object):
+    PENDING = 'pending'
+    RUNNING = 'running'
+    FAILED = 'failed'
+    SUCCEEDED = 'succeeded'
+
+
+class PodLauncher(LoggingMixin):
+    def __init__(self, kube_client=None):
+        self._client = kube_client or get_kube_client()
+        self._watch = watch.Watch()
+
+    def run_pod_async(self, pod):
+        req = self.kube_req_factory.create(pod)
+        self.log.debug('Pod Creation Request: \n{}'.format(json.dumps(req, indent=2)))
+        try:
+            resp = self._client.create_namespaced_pod(body=req, namespace=pod.namespace)
+            self.log.debug('Pod Creation Response: {}'.format(resp))
+        except ApiException:
+            self.log.exception('Exception when attempting to create Namespaced Pod.')
+            raise
+        return resp
+
+    def run_pod(self, pod):
+        # type: (Pod) -> State
+        """
+        Launches the pod synchronously and waits for completion.
+        """
+        resp = self.run_pod_async(pod)
+        if resp.status.start_time is None:
+            self.log.debug('Pod not yet started')
+
+        final_status = self._monitor_pod(pod)
+        return final_status
+
+    def _monitor_pod(self, pod):
+        # type: (Pod) -> State
+
+        while self.pod_is_running(pod):
+            self.log.info("Pod {} has state {}".format(pod.name, State.RUNNING))
+            time.sleep(2)
+        return self.read_pod(pod)
+
+        # for event in self._watch.stream(self.read_pod(pod), pod.namespace):
```

Comment:
Reminder to delete this. I think polling seems like a good solution for the operator (not the scheduler).